File: 5th Semester/Mobile/UI Native Swift/View/CarCard.swift
//
//  CarCard.swift
//  Rental
//
//  Created by Mircea Maierean on 07.11.2024.
//

import SwiftUI

struct CarCard: View {
    var car: Car

    var body: some View {
        VStack(alignment: .leading) {
            if let imageName = car.photo {
                Image(imageName)
                    .resizable()
                    .scaledToFill()
                    .frame(height: 150)
                    .clipped()
                    .cornerRadius(8)
            } else {
                Rectangle()
                    .fill(Color.gray)
                    .frame(height: 150)
                    .cornerRadius(8)
            }

            VStack(alignment: .leading, spacing: 8) {
                Text("\(car.make) \(car.model)")
                    .font(.title3)
                    .fontWeight(.semibold)

                Text("Price per day: $\(car.pricePerDay)")
                    .font(.subheadline)
                    .foregroundColor(.gray)

                Text("Color: \(car.color)")
                    .font(.subheadline)

                Text("Body type: \(car.bodyType)")
                    .font(.subheadline)

                Text("Year: \(String(format: "%d", car.productionYear))")
                    .font(.subheadline)
            }
            .padding(.top, 8)
            .padding([.leading, .trailing], 4)
            .frame(maxWidth: .infinity, alignment: .leading)
        }
        .padding()
        .background(Color.white)
        .cornerRadius(12)
        .shadow(radius: 5)
    }
}

File: 5th Semester/mobile/localdb/rental/lib/views/car_card.dart
import 'package:flutter/material.dart';
import '../models/car.dart';

class CarCard extends StatelessWidget {
  final Car car;

  const CarCard({required this.car, super.key});

  @override
  Widget build(BuildContext context) {
    return Padding(
      padding: const EdgeInsets.all(8.0),
      child: Container(
        decoration: BoxDecoration(
          color: Colors.white,
          borderRadius: BorderRadius.circular(12),
          boxShadow: [
            BoxShadow(
              color: Colors.black26,
              blurRadius: 5,
              offset: Offset(0, 2),
            ),
          ],
        ),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            car.photoPath != null
                ? ClipRRect(
                    borderRadius:
                        BorderRadius.vertical(top: Radius.circular(12)),
                    child: Image.asset(
                      car.photoPath!,
                      height: 150,
                      width: double.infinity,
                      fit: BoxFit.cover,
                    ),
                  )
                : Container(
                    height: 150,
                    decoration: BoxDecoration(
                      color: Colors.grey,
                      borderRadius:
                          BorderRadius.vertical(top: Radius.circular(12)),
                    ),
                  ),
            Padding(
              padding: const EdgeInsets.all(8.0),
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  Text(
                    '${car.make} ${car.model}',
                    style: Theme.of(context)
                        .textTheme
                        .titleMedium
                        ?.copyWith(fontWeight: FontWeight.w600),
                  ),
                  const SizedBox(height: 4),
                  Text(
                    'Price per day: \$${car.pricePerDay}',
                    style: Theme.of(context)
                        .textTheme
                        .bodySmall
                        ?.copyWith(color: Colors.grey),
                  ),
                  const SizedBox(height: 4),
                  Text(
                    'Color: ${car.color}',
                    style: Theme.of(context).textTheme.bodySmall,
                  ),
                  Text(
                    'Body type: ${car.bodyType}',
                    style: Theme.of(context).textTheme.bodySmall,
                  ),
                  Text(
                    'Year: ${car.productionYear}',
                    style: Theme.of(context).textTheme.bodySmall,
                  ),
                ],
              ),
            ),
          ],
        ),
      ),
    );
  }
}
